cli/minipresto/cli.py

```diff
@@ -25,98 +48,109 @@ def __init__(self):
+
+        Methods
+        -------
+        - `log()`: Log messages to the user's console.
+        - `prompt_msg()`: Prints a prompt message and returns the user's input.
+        - `get_config()`: Returns an instantiated ConfigParser object from the
+          Minipresto user config. file.
+        - `get_config_value()`: Returns a value from the config if present.
         """
 
+        # Warnings
+        self._initial_warnings = []
+
         # Verbose logging
         self.verbose = False
 
         # Paths
         self.user_home_dir = os.path.expanduser("~")
-        self.minipresto_user_dir = self.handle_minipresto_user_dir()
-        self.config_file = os.path.join(self.minipresto_user_dir, "minipresto.cfg")
+        self.minipresto_user_dir = self._handle_minipresto_user_dir()
+        self.config_file = self._get_config_file()
         self.snapshot_dir = os.path.join(self.minipresto_user_dir, "snapshots")
 
         # Points to the directory containing minipresto library. Library
         # consists of modules, snapshots, and module parent files
-        self.minipresto_lib_dir = self.get_minipresto_lib_dir()
-
-    def log(self, *args):
-        """Logs a message."""
-
-        for arg in args:
-            arg = self.transform_log_msg(arg)
-            if not arg: return
-            click.echo(
-                click.style(
-                    f"[i]  {click.style(arg, fg='cyan', bold=False)}",
-                    fg="cyan",
-                    bold=True,
-                )
-            )
+        self.minipresto_lib_dir = self._get_minipresto_lib_dir()
 
-    def log_warn(self, *args):
-        """Logs a warning message."""
-
-        for arg in args:
-            arg = self.transform_log_msg(arg)
-            if not arg: return
-            click.echo(
-                click.style(
-                    f"[w]  {click.style(arg, fg='yellow', bold=False)}",
-                    fg="yellow",
-                    bold=True,
-                )
-            )
+        # Docker clients
+        self.docker_client, self.api_client = self._get_docker_clients()
 
-    def log_err(self, *args):
-        """Logs an error message."""
+    def log(self, *args, level=LogLevel().info):
+        """
+        Logs messages to the user's console. Defaults to 'info' log level.
 
-        for arg in args:
-            arg = self.transform_log_msg(arg)
-            if not arg: return
-            click.echo(
-                click.style(f"[e]  {click.style(arg, fg='red')}", fg="red", bold=True,)
-            )
+        Parameters
+        ----------
+        - `*args`: Messages to log.
+        - `level`: The level of the log message.
+        """
+
+        # Skip verbose messages unless verbose mode is enabled
+        if not self.verbose and level == LogLevel().verbose:
+            return
 
-    def vlog(self, *args):
-        """Logs a message only if verbose logging is enabled."""
+        for msg in args:
```

Comment:
For reference, carriage returns really tripped everything up. It seemed generally reasonable to do this for consistency.